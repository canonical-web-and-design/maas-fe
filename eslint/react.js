module.exports = {
  plugins: ["react"],
  rules: {
    "react/forbid-component-props": [
      "error",
      {
        forbid: [
          {
            propName: "data-test",
            message: "Use `data-testid` instead of `data-test` attribute",
          },
        ],
      },
    ],
    "react/forbid-dom-props": [
      "error",
      {
        forbid: [
          {
            propName: "data-test",
            message: "Use `data-testid` instead of `data-test` attribute",
          },
        ],
      },
    ],
  },
};
