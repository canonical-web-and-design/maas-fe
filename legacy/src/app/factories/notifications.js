/* Copyright 2017 Canonical Ltd.  This software is licensed under the
 * GNU Affero General Public License version 3 (see the file LICENSE).
 *
 * MAAS Notifications Manager.
 *
 * Manages notifications in the browser. Uses RegionConnection to load
 * notifications, await new, updated, and deleted notifications, and to
 * dismiss them.
 */
import angular from "angular";

function NotificationsManager(RegionConnection, Manager) {
  function NotificationsManager() {
    Manager.call(this);

    this._pk = "id";
    this._handler = "notification";

    // Listen for notify events for the notification object.
    RegionConnection.registerNotifier(
      "notification",
      angular.bind(this, this.onNotify)
    );
  }

  NotificationsManager.prototype = new Manager();
  NotificationsManager.prototype.dismiss = function (notification) {
    return RegionConnection.callMethod("notification.dismiss", {
      id: notification.id,
    });
  };
  NotificationsManager.prototype.getItems = function () {
    return this._items.filter(({ ident }) => ident !== "release_notification");
  };

  return new NotificationsManager();
}

NotificationsManager.$inject = ["RegionConnection", "Manager"];

export default NotificationsManager;
