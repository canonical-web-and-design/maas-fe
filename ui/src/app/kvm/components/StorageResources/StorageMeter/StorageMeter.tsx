import PodMeter from "app/kvm/components/PodMeter";
import StoragePopover from "app/kvm/components/StorageColumn/StoragePopover";
import type { PodStoragePool } from "app/store/pod/types";
import { formatBytes } from "app/utils";

type Props = {
  pool: PodStoragePool;
};

const StorageMeter = ({ pool }: Props): JSX.Element | null => {
  const total = formatBytes(pool.total, "B");
  const allocated = formatBytes(pool.used, "B", {
    convertTo: total.unit,
  });
  const free = formatBytes(pool.total - pool.used, "B", {
    convertTo: total.unit,
  });

  return (
    <div className="u-width--full">
      <StoragePopover pools={[pool]}>
        <PodMeter
          allocated={allocated.value}
          className="storage-meter__meter"
          free={free.value}
          unit={total.unit}
        />
      </StoragePopover>
    </div>
  );
};

export default StorageMeter;
