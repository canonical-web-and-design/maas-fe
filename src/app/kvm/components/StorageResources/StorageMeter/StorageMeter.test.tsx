import { render, screen } from "@testing-library/react";

import StorageMeter from "./StorageMeter";

import * as factory from "@/testing/factories";

const pools = {
  "pool-1": factory.podStoragePoolResource({
    allocated_other: 1,
    allocated_tracked: 2,
    backend: "zfs",
    name: "pool-1",
    path: "/path1",
    total: 3,
  }),
};

describe("StorageMeter", () => {
  it("renders", () => {
    render(<StorageMeter pools={pools} />);

    expect(screen.getByText("Allocated")).toBeInTheDocument();
  });

  it("does not render if more than one pool present", () => {
    render(
      <StorageMeter
        pools={{ ...pools, "pool-2": factory.podStoragePoolResource() }}
      />
    );

    expect(screen.queryByText("Allocated")).not.toBeInTheDocument();
  });
});
