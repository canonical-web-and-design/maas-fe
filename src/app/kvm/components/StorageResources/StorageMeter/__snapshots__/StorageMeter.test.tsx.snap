// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StorageMeter renders 1`] = `
<div
  className="u-width--full"
>
  <StoragePopover
    pools={
      Object {
        "pool-1": Object {
          "allocated_other": 1,
          "allocated_tracked": 2,
          "backend": "zfs",
          "id": "abc123",
          "name": "pool-1",
          "path": "/path1",
          "total": 3,
        },
      }
    }
  >
    <KVMResourceMeter
      allocated={2}
      detailed={true}
      free={0}
      other={1}
      unit="B"
    />
  </StoragePopover>
</div>
`;
